sap/cli/startrfc.py

```diff
@@ -89,10 +94,39 @@ def startrfc(connection, args):
         console.printerr(f'{args.RFC_FUNCTION_MODULE} failed:')
         console.printerr(str(ex))
         return 1
-    else:
-        console.printout(FORMATTERS[args.output](resp))
+
+    # rfc call passed, it is time for analysis of the returned response
+
+    # if bapi checker  is enabled
+    if args.result_checker == RESULT_CHECKER_BAPI:
+        # first check of the response to be sure it fits to BAPI response structure
+        try:
+            return_value = resp['RETURN']
+        except KeyError:
+            console.printerr(f"It was requested to evaluate response from {args.RFC_FUNCTION_MODULE} "
+                             "as bapi result, but response does not contain required key RETURN")
+            return 1
```

Comment:
I know that someone made an error but I would not punish them so much and I would fall back to "raw" to at least give them all results (still with exit code 1). We are going to use it in Jenkins builds, it happens a lot that we call something wrong and more logs you have the better.

Reply:
Yes, you are right, thanks for this catch, I'll add raw output for all cases when BAPI parsing fails.